pkg/sfu/downtrack.go

```diff
@@ -1091,6 +1096,27 @@ func (d *DownTrack) DebugInfo() map[string]interface{} {
 	}
 }
 
-func (d *DownTrack) GetConnectionScore() float64 {
+func (d *DownTrack) GetConnectionScore() float32 {
 	return d.connectionStats.GetScore()
 }
+
+func (d *DownTrack) getTrackStats() map[uint32]*buffer.StreamStatsWithLayers {
+	d.statsLock.RLock()
+	defer d.statsLock.RUnlock()
+
+	stats := make(map[uint32]*buffer.StreamStatsWithLayers, 1)
+
+	layers := make(map[int]buffer.LayerStats)
+	layers[0] = buffer.LayerStats{
+		TotalPackets: d.stats.TotalPrimaryPackets + d.stats.TotalRetransmitPackets + d.stats.TotalPaddingPackets,
+		TotalBytes:   d.stats.TotalPrimaryBytes + d.stats.TotalRetransmitBytes + d.stats.TotalPaddingBytes,
+		TotalFrames:  d.stats.TotalFrames,
+	}
+
+	stats[d.ssrc] = &buffer.StreamStatsWithLayers{
+		StreamStats: d.stats,
```

Comment:
DownTrack would always have a single layer, would this still be used with the conversion above? `cs.params.CodecType == webrtc.RTPCodecTypeVideo && (len(streams) > 1 || len(stream.Layers) > 1)`

Reply:
Be able to surface layer information flexibly. Covering cases of non-simulcast, simulcast and SVC. Each entity (could be `sfu.Buffer` or `sfu.WebRTCReceiver` or `sfu.DownTrack` can report whatever layers they want to report. Analytics uses the above rule to decide whether to include layers in the protocol message.

Examples of how it can be used
- `sfu.Buffer` is per RTP stream. It currently reports one layer. `sfu.Receiver` has to do some munging (just the layer value) for simulcast tracks as `sfu.Buffer` does not itself know which layer it is in.
- For SVC tracks, `sfu.Buffer` itself would maintain layer stats and it will report multiple layers. In that case `sfu.Receiver` can pass that through directly to `ConnectionStats`
- `sfu.DownTrack` currently reports only one layer. But, I was thinking of reporting multiple layers at some point from there too. It will be a nice way to see the switch points visually. So, although single stream, `sfu.DownTrack` knows which layer it is forwarding and can maintain that information and populate layers. That can apply equally when forwarding simulcast or SVC.

`ConnectionStats` uses the above rule to decide whether to include layers or not. It is included if simulcast (number of streams will be more than 1) or SVC (number of layers in a stream is more than 1). In the future, if we decide to send layer information from down track (third point above), we can just make the change in `sfu.DownTrack` to report more layers and this piece of code will report it to analytics.

Currently, the following cases will not report layer to analytics
- Audio
- Non-simulcast video as there is only one stream and also only one layer in that stream
- DownTrack video as there is only one stream and currently we are reporting only one layer in that stream although it is multiplexing.